samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/author/AssessmentSettingsBean.java

```diff
@@ -1773,7 +1774,7 @@ public void saveEditedExtendedTime() {
       Integer oldIndex = null;
       // is there a more efficient way?
       for(ExtendedTime entry : extendedTimes) {
-        if(entry.getId().equals(this.extendedTime.getId())) {
+        if(ObjectUtils.equals(entry.getId(), this.extendedTime.getId())) {
```

Comment:
Fixing bugs++.

The point of this loop is to go through the extendedTime entries to find the old edited entry, and replacing it with the new edit. If both IDs are null, one can't be 100% sure that you've found the old entry (what if you have two entries with a null id?).

I think something like the following might be a better solution:

```diff --git a/samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/author/AssessmentSettingsBean.java b/samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/author/AssessmentSettingsBean.java
index 1bc8e62024..69d58631ae 100644
--- a/samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/author/AssessmentSettingsBean.java
+++ b/samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/author/AssessmentSettingsBean.java
@@ -1765,20 +1765,13 @@ public class AssessmentSettingsBean
     public void editExtendedTime() {
       this.editingExtendedTime = true;
       this.extendedTime = new ExtendedTime(this.transitoryExtendedTime);
-      this.transitoryExtendedTime = null;
     }
 
     public void saveEditedExtendedTime() {
       this.editingExtendedTime = false;
-      Integer oldIndex = null;
-      // is there a more efficient way?
-      for(ExtendedTime entry : extendedTimes) {
-        if(entry.getId().equals(this.extendedTime.getId())) {
-          oldIndex = this.extendedTimes.indexOf(entry);
-          break;
-        }
-      }
+      Integer oldIndex = extendedTimes.indexOf(this.transitoryExtendedTime);
       this.extendedTimes.set(oldIndex, this.extendedTime);
       resetExtendedTime();
     }
```

this.transitoryExtendedTime (tET) is just used to pass the desired extended time entry for the delete and edit functionality. When the delete or edit functionality is used, tET is set to the ExtendedTime entry you wish to delete / edit, so you can use this Object to find the old extendedTime entry. You could nullify tET at the end of saveEditedExtendedTime, but that is not necessary.